src/stats/archive/calculator.rs

```diff
@@ -76,7 +114,7 @@ impl ArchiveStats {
     pub fn max_batch_size_per_caster(
         &self,
         slots_in_epoch: u32,
-    ) -> Result<BTreeMap<String, usize>, ArchiveCalculatorError> {
+    ) -> Result<Vec<CasterBatchSize>, ArchiveCalculatorError> {
```

Comment:
same here, may `caster_id` appear twice in the `Vec` with different meanings? Otherwise a map is more idiomatic